sync/class.jetpack-sync-actions.php

```diff
@@ -315,6 +308,56 @@ static function initialize_sender() {
 		// bind the sending process
 		add_filter( 'jetpack_sync_send_data', array( __CLASS__, 'send_data' ), 10, 4 );
 	}
+
+	static function sanitize_filtered_sync_cron_schedule( $schedule ) {
+		$schedule = sanitize_key( $schedule );
+		$schedules = wp_get_schedules();
+
+		// Make sure that the schedule has actually been registered using the `cron_intervals` filter.
+		if ( isset( $schedules[ $schedule ] ) ) {
+			return $schedule;
+		}
+
+		return self::DEFAULT_SYNC_CRON_INTERVAL;
+	}
+
+	static function maybe_schedule_sync_cron( $schedule, $hook ) {
+		if ( ! $hook ) {
```

Comment:
Perhaps. I added it in there as a safeguard since I also refactored the scheduling functionality into a static method. I did that for testing purposes, specifically if we decide to test just that method. But, now that we're exposing it, I think we should also make sure that others don't send us junk.